src/pubsub/ua_pubsub_config.c

```diff
@@ -154,17 +154,19 @@ setConnectionPublisherId(const UA_PubSubConnectionDataType *src,
                          UA_PubSubConnectionConfig *dst) {
     if(src->publisherId.type == &UA_TYPES[UA_TYPES_STRING]) {
         dst->publisherIdType = UA_PUBSUB_PUBLISHERID_STRING;
-        dst->publisherId.string = *(UA_String*)src->publisherId.data;
-    } else if(src->publisherId.type == &UA_TYPES[UA_TYPES_BYTE] ||
-              src->publisherId.type == &UA_TYPES[UA_TYPES_UINT16] ||
-              src->publisherId.type == &UA_TYPES[UA_TYPES_UINT32]) {
-        dst->publisherIdType = UA_PUBSUB_PUBLISHERID_NUMERIC;
-        dst->publisherId.numeric =  *(UA_UInt32*)src->publisherId.data;
+        dst->publisherId.publisherIdString = *(UA_String*)src->publisherId.data;
```

Comment:
Won't this require a deep copy of the string?

Reply:
It shouldn't because `UA_Server_addPubSubConnection() -> UA_PubSubConnectionConfig_copy()` creates the deep copy

I have tried to write a unit-test (see last commit), but I couldn't get it working
I don't have a PubSub binary config file, so I thought I could create UA_PubSubConfigurationDataType structure with unittest, then encode it to bytestring and then use it to simulate PubSub file - already loaded bytestring

But `UA_PubSubManager_loadPubSubConfigFromByteString` fails at decoding and I don't know why

When I try to use `encodePubSubConfiguration` https://github.com/open62541/open62541/blob/c6e4e028a740b50c49a6dd5987ee8e61a52953c9/src/pubsub/ua_pubsub_config.c#L834 to create bytestring of PubSubConfigurationDataType it does not compile because functions `UA_ExtensionObject_calcSizeBinary` and `UA_ExtensionObject_encodeBinary` are used, which are not available anymore I think

So I'm confused why it compiles in the pipeline and how to use it correctly.

Reply:
It looks like `UA_ENABLE_PUBSUB_FILE_CONFIG` is not enabled at any test within https://github.com/open62541/open62541/blob/master/tools/ci.sh

which I think configures the tests right? because the previous travis scripts are obsolete?

Reply:
Sorry, I was totally confused

I wanted to test PubSub configuration with a file, but I don't have a file, so I created PubSubConfigurationDataType structure in unit-test and tried to encode it to ByteString.
I wanted to use UA_encodeBinary but it didn't work and then I confused a lot of things

But I think it should work now and I have added the test for PubSub file configuration with PublisherId strings